### DISCUSSION TITLE
Any way to spawn a new tab with a command and the execution?

### DISCUSSION THREAD

**Discussion by carlos-gtz** [wezterm/wezterm#2202] [Q&A] [answered]
I'm using `wezterm cli spawn -- ls -la` but wezterm new tab closes after the command is finished. I want it to keep it alive. This is because I would like to create some commands for using in [Raycast](https://www.raycast.com/) 🤔 

I tried using the send-text and a new line `\n` but with no success. It just prints the command but it doesn't execute it.

**Comment by wez** [comment, marked as answer]
Two suggestions:

* Pad out the execution with a long sleep: `wezterm cli spawn -- bash -c "ls -la ; sleep 86400"` to keep the pane alive for 1 day in this case.
* Set [exit_behavior](https://wezfurlong.org/wezterm/config/lua/config/exit_behavior.html) = `"Hold"` to have wezterm keep every pane open until you explicitly close the pane.

**Reply by carlos-gtz** [reply]
Thanks, this really helps me a lot :D
Any way to be able to keep using that pane? The execution of the command will wait but I want it to be finished and then start using the pane. An example would be like "opening a new pane in a different cwd with a compile script. After finish, then the user can use the pane and keep the same cwd from where the compilation occurred"

**Reply by wez** [reply]
You may wish to do something like this to spawn a shell and send the initial command:

```
pane_id=$(wezterm cli spawn)
echo -e "ls -la\n" | wezterm cli send-text --no-paste --pane-id $pane_id
```

**Reply by carlos-gtz** [reply]
Thanks a lot :)
